### DISCUSSION TITLE
Viva connections card notification subscription socket io question - getAllMessages  teams chats and channels

### DISCUSSION THREAD

**Discussion by spoCloud** [SharePoint/sp-dev-docs#7980] [Q&A]
hi all,

I'm trying to develop a Viva connections card that subscribes to /teams/getAllMessages   and /chats/getAllMessages  notifications and show change notifications on the card.    But there is no . . ./subscriptions/socketIO   for chats and channels.   I'm not sure how the socket.io works in this case.  Anyone got a good tutorial or sample code?

Thanks

**Comment by AJIXuMuK** [comment]
@spoCloud - I think you could create server-side subscription for the messages with app-only permissions.
Then you could socket.io to communicate between you server-side app and your card.
Another option is just ping the MS Graph once in a while and get the [delta](https://docs.microsoft.com/en-us/graph/api/chatmessage-delta?view=graph-rest-1.0&tabs=http)

**Reply by spoCloud** [reply]
@AJIXuMuK  - thanks for the suggestions.  I was hoping for little more details on how exactly to implement the server side subscription.  I can create an Azure function with http trigger to receive graph notifications.  But I'm not sure how to add the socket code in the server code ( azure function ).     I'm trying out the delta queries.    Thanks !
